docs/types.rst

```diff
@@ -0,0 +1,318 @@
+
+
+.. cfy:node:: cloudify.aws.nodes.Instance
+
+  The public key which is set for the server needs to match the private key name in your AWS account. The public key may be set in a number of ways:
+
+    * By connecting the instance node to a keypair node using the ``cloudify.aws.relationships.instance_connected_to_keypair`` relationship.
+    * By setting it explicitly in the ``key_name`` key under the ``parameters`` property.
+    * If the agent's keypair information is set in the provider context, the agents' keypair will serve as the default public key to be used if it was not specified otherwise.
+
+
+If the server is to have an agent installed on it, it should use the agents security group. If you are using a manager bootstrapped with the standard aws-manager-blueprint, there is a provider context dictionary on the manager that provides this value to the plugin. You can also use other security groups by:
+  * ``security_groups``: list of security group names.
+  * ``security_group_ids``: a list of security group IDs.
+
+If you want to specify the ``availability_zone`` for your instance, you must use the ``placement`` key.
+
+The create function also sets ``reservation_id`` attribute. For information, see [here](http://boto.readthedocs.org/en/latest/ref/ec2.html#boto.ec2.instance.Reservation)
+
+Four additional ``runtime_properties`` are available on node instances of this type once the ``cloudify.interfaces.lifecycle.start`` operation succeeds:
+
+  * ``ip`` the instance's private IP.
+  * ``private_dns_name`` the instance's private FQDN in Amazon.
+  * ``public_dns_name`` the instances's public FQDN in Amazon.
+  * ``public_ip_address`` the instance's public IP address.
+
+.. tip::
+    If you want to use the instance in VPC, then you need to connect this to a Subnet using the `cloudify.aws.relationships.instance_contained_in_subnet` relationship.
+
+.. cfy:node:: cloudify.aws.nodes.WindowsInstance
+
+    Use this type when working with a Windows server.
+    It has the same properties and operations-mapping as
+    :cfy:node:`cloudify.aws.nodes.Instance`,
+    yet it overrides some of the agent and plugin installations operations-mapping derived from
+    the :cfy:node:`cloudify.nodes.Compute` type.
+
+    Additionally, the default value for the ``use_password`` property
+    is overridden for this type, and is set to ``true``.
+    In this case, the password of the windows server will be retrieved,
+    decrypted and put under the ``password`` runtime property of this node instance.
+
+
+.. cfy:node:: cloudify.aws.nodes.KeyPair
+
+
+.. cfy:node:: cloudify.aws.nodes.SecurityGroup
+
+.. note::
+    If you want to create a security group in a VPC,
+    you need to connect it to a VPC using the
+    :cfy:rel:`cloudify.aws.relationships.security_group_contained_in_vpc`
+    relationship.
+
+
+.. cfy:node:: cloudify.aws.nodes.ElasticIP
+
+.. note::
+    the actual IP is available via the ``aws_resource_id`` runtime-property.
```

Comment:
I'm sure that underlying this diversion of the norm is some very interesting reason, but not interesting enough to warrant the search for it.